week2/search.py

```diff
@@ -57,20 +57,29 @@ def process_filters(filters_input):
 
 @bp.route('/autocomplete', methods=['GET'])
 def autocomplete():
+    opensearch = get_opensearch()
     results = {}
     if request.method == 'GET':  # a query has been submitted
-        prefix = request.args.get("prefix")
-        print(f"Prefix: {prefix}")
+        prefix = request.args.get("prefix")  # not sure what should be default
         if prefix is not None:
-            type = request.args.get("type", "queries") # If type == queries, this is an autocomplete request, else if products, it's an instant search request.
+            type = request.args.get("type", "queries")
             ##### W2, L3, S1
             search_response = None
             print("TODO: implement autocomplete AND instant search")
-            if (search_response and search_response['suggest']['autocomplete'] and search_response['suggest']['autocomplete'][0]['length'] > 0): # just a query response
-                results = search_response['suggest']['autocomplete'][0]['options']
-    print(f"Results: {results}")
+            index_name = 'bbuy_queries' if type == 'queries' else 'bbuy_products'
+            # this is an autocomplete request, 
+            if type == 'queries':
+                query_obj = qu.get_autocomplete_query(prefix=request.args.get('prefix'))
+                search_response = opensearch.search(body=query_obj, index=index_name)
+                if (search_response and search_response['suggest']['autocomplete'] and search_response['suggest']['autocomplete'][0]['length'] > 0): # just a query response
+                    results = search_response['suggest']['autocomplete'][0]['options']
+                    print(f"Results: {results}")
+            else: # if products, it's an instant search request.
+                pass
```

Comment:
Instant search in our case would mean we should still search, it should just go against the products index.